cluster.yml

```diff
@@ -40,7 +40,27 @@
     - { role: kubespray-defaults }
     - { role: bootstrap-os, tags: bootstrap-os}
 
+- name: Gather facts
+  hosts: all
+  gather_facts: False
+  tasks:
+    - name: Gather minimal facts
+      setup:
+        gather_subset: '!all'
+
+    - name: Gather necessary facts
+      setup:
+        gather_subset: '!all,!min,network,hardware'
+        filter: "{{ item }}"
+      loop:
+        - ansible_distribution_major_version
```

Comment:
This looping will take 8x longer than just gathering all facts

Reply:
why ?
the problem, as I understand it, is that on a node with running kubernetes and lots of pods, the size of facts is 30 megabytes, and the ansible loads this cache very slowly for each play

so with this task fact cache size is 10kb

Reply:
Ansible is slow when handling the large number of facts. This is a one time operation that ensures that the hostvars are kept small, speeding up all the other tasks.

Reply:
and it was 3MB -> 30kB in our case :-)

Reply:
The number of loop iterations here is constant and the gathers will occur across all available forks in parallel so I don't have a problem with it.

A previous method of looping through hosts to gather facts with delegation had n^2 iterations which were not ran in parallel. That method could easily take 8 times longer than a gather all so I understand the apprehension with a loop. However, this patch does not have that problem since each setup iteration works on as many nodes as allowed by forks.